src/api/app/controllers/webui/projects/maintenance_incidents_controller.rb

```diff
@@ -0,0 +1,50 @@
+module Webui
+  module Projects
+    class MaintenanceIncidentsController < WebuiController
+      before_action :set_project
+      before_action :lockout_spiders, only: [:index]
+      before_action :require_login, only: [:create, :create_request]
+
+      after_action :verify_authorized, except: [:index, :create, :create_request]
+
+      def index
+        @incidents = @project.maintenance_incidents
+      end
+
+      def create
+        authorize @project, :update?
```

Comment:
I thinkyou still need `project = Project.get_by_name(params[:ns])`, is different than the `set_project`